### ISSUE TITLE
willBeRetried: false for all scenarios regardless of retries count

### ISSUE THREAD

**Issue by WTK** [badeball/cypress-cucumber-preprocessor#849]
### Current behavior

cucumber messages always have `willBeRetried: false` for test cases despite them being retried.

I think as a side effect (or direct effect?) the html report generated has no screenshots attached. They're attached when retries == 0, but have no screenshots otherwise.

### Desired behavior

Messages have `willBeRetried: true` if test case will be retried and `false` otherwise.

### Test code to reproduce

No real code - just have a test that is failing and set retries number in cypress to be greater than zero. Observe resulting messages - they all have aforementioned property set to false.

### Versions

* **Cypress version**: 10.8.0
* **Preprocessor version**: 13.0.2
* **Node version**: 14.14

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by WTK** [comment]
It's easy to fix by running this within `it` handler, setting state in properties (retrieved via `retrieveInternalSpecProperties`) and then read state in the `afterEach` right when `testCaseFinished` message is pushed to the stack.

```
it(scenarioName, suiteOptions, function() {
    let attempt = cy.state('runnable')._currentRetry
    let retries = cy.state('runnable')._retries
    let isLastAttempt = attempt === retries
    const properties = retrieveInternalSpecProperties();
    properties._retriesState = {
        attempt,
        retries,
        isLastAttempt,
    };
})

afterEach(function() {
    messages.push({
        testCaseFinished: {
            testCaseStartedId,
            timestamp: endTimestamp,
            willBeRetried: !properties._retriesState.isLastAttempt,
        },
    });
})
```

With that said it doesn't help currently with html output (it still has no screenshots when retries > 0).

**Comment by badeball** [comment]
Feel free to open up a PR on this. I think you can avoid mutating the internal properties though and rather just calculate `isLastAttempt` in the `afterEach(..)` block. Also, a change in the messages output should be accompanied by at least one test (in this case it's fine to add a scenario in `features/issues/849.feature`).

**Comment by WTK** [comment]
In my testing those internal properties of cypress were unavailable in the afterEach hook. I'm guessing they clear this information sometime before that.

**Comment by badeball** [comment]
This has been fixed with v16.0.0.
